File: examples/modulith-clinical-tool-jpa/src/test/java/io/zenwave360/example/clinicaltool/modules/masterdata/config/ServicesInMemoryConfig.java
package io.zenwave360.example.clinicaltool.modules.masterdata.config;

import io.zenwave360.example.clinicaltool.common.TestDataLoader;
import io.zenwave360.example.clinicaltool.modules.masterdata.*;
import io.zenwave360.example.clinicaltool.modules.masterdata.domain.*;
import java.util.List;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.context.annotation.Profile;

/**
 * Services InMemory Config. It can be used standalone or with @SpringBootTest.
 */
@Configuration
@Profile("in-memory")
public class ServicesInMemoryConfig extends RepositoriesInMemoryConfig {

    protected final MasterDataServiceImpl masterDataService = new MasterDataServiceImpl(masterDataRepository());

    @Bean
    public <T extends MasterDataService> T masterDataService() {
        return (T) masterDataService;
    }

    static List<MasterData> _masterData;

    public void reloadTestData() {

        var testDataLoader = new TestDataLoader(List.of(MasterData.class, MasterDataTranslation.class));
        var masterData =
                _masterData != null ? _masterData : testDataLoader.loadCollectionTestDataAsObjects(MasterData.class);
        masterDataRepository().deleteAll();
        masterDataRepository().saveAll(masterData);
    }
}

File: examples/modulith-klinical-tool-jpa/src/test/kotlin/io/zenwave360/example/clinicaltool/modules/masterdata/config/ServicesInMemoryConfig.kt
package io.zenwave360.example.clinicaltool.modules.masterdata.config

import io.zenwave360.example.clinicaltool.modules.masterdata.*
import io.zenwave360.example.clinicaltool.modules.masterdata.domain.*
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.context.annotation.Profile

/** Services InMemory Config. It can be used standalone or with @SpringBootTest. */
@Configuration
@Profile("in-memory")
open class ServicesInMemoryConfig : RepositoriesInMemoryConfig() {

    protected val masterDataService = MasterDataServiceImpl(masterDataRepository())

    @Bean
    open fun <T : MasterDataService> masterDataService(): T {
        return masterDataService as T
    }

    companion object {
        var _masterData: List<MasterData>? = null
    }

    fun reloadTestData() {

        val testDataLoader = TestDataLoader(listOf(MasterData::class.java, MasterDataTranslation::class.java))
        val masterData = _masterData ?: testDataLoader.loadCollectionTestDataAsObjects(MasterData::class.java)
        masterDataRepository().deleteAll()
        masterDataRepository().saveAll(masterData)
    }
}
